pkg/trace/writer/stats.go

```diff
@@ -36,25 +37,31 @@ const (
 
 // StatsWriter ingests stats buckets and flushes them to the API.
 type StatsWriter struct {
-	in       <-chan []stats.Bucket
-	hostname string
-	env      string
-	senders  []*sender
-	stop     chan struct{}
-	stats    *info.StatsWriterInfo
+	in        <-chan []stats.Bucket
+	hostname  string
+	env       string
+	senders   []*sender
+	stop      chan struct{}
+	flushChan chan struct{}
+	wg        sync.WaitGroup
+	stats     *info.StatsWriterInfo
+	payloads  []*stats.Payload // payloads buffered for sync mode
+	syncMode  bool
```

Comment:
A small nit for making more sense of this. Do you mind if we add an empty line between these two and the rest? They are like a separate category exclusively for sync. Also reversing the order might make sense. I mean something like:
```suggestion

	syncMode  bool             // sync mode enabled -- writer only flushes when FlushSync is called.
	payloads  []*stats.Payload // payloads buffered for sync mode
```